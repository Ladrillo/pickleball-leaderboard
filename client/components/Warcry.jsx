import React from 'react';


const challengeStyle = {
    fontFamily: 'Luckiest Guy',
    color: '#5213D4',
    textAlign: 'center',
    fontSize: 24
};



const Warcry = props => {

    const potentialVictim = (
        !props.me.stats.locked.id &&
        !props.player.stats.locked.id &&
        props.isChallengeable
    );

    if (potentialVictim) {
        return (
            <div
                style   = { challengeStyle }
                onClick = { () => props.challengeHandler(props.me, props.player) }>
                challenge {props.player.displayName.split(' ')[0]}!
            </div>
        );
    }
    return null;
};


export default Warcry;