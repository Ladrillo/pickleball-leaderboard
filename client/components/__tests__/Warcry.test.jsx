import React from 'react';
import renderer from 'react-test-renderer';
import Warcry from '../Warcry';


describe(`Warcry`, () => {

    it(`Renders warcry if the player is challengeable`, () => {
        const component = renderer.create(
            <Warcry
                player={{
                    "_id": "580511cdb9edbe0010bd66e8",
                    "displayName": "Darien Hess",
                    "__v": 0,
                    "stats": {
                        "locked": { "id": "" },
                        "score": 0,
                        "joined": "2016-10-17T18:00:45.253Z"
                    },
                    "google": { "id": "102827209431078726602" }
                }}
                me={{
                    "_id": "5804518e09573008c8319060",
                    "displayName": "Gabriel Cabrejas",
                    "__v": 0,
                    "stats": {
                        "locked": { "id": "" },
                        "score": 15,
                        "joined": "2016-10-17T04:20:30.942Z"
                    },
                    "google": { "id": "100217230618184900301" }
                }}
                isChallengeable={true} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it(`returns null if the player is not challengeable`, () => {
        const component = renderer.create(
            <Warcry
                player={{
                    "_id": "580511cdb9edbe0010bd66e8",
                    "displayName": "Darien Hess",
                    "__v": 0,
                    "stats": {
                        "locked": { "id": "" },
                        "score": 0,
                        "joined": "2016-10-17T18:00:45.253Z"
                    },
                    "google": { "id": "102827209431078726602" }
                }}
                me={{
                    "_id": "5804518e09573008c8319060",
                    "displayName": "Gabriel Cabrejas",
                    "__v": 0,
                    "stats": {
                        "locked": { "id": "" },
                        "score": 15,
                        "joined": "2016-10-17T04:20:30.942Z"
                    },
                    "google": { "id": "100217230618184900301" }
                }}
                isChallengeable={false} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});