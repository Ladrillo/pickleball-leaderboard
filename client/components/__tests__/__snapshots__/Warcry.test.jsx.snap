exports[`Warcry Renders warcry if the player is challengeable 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "color": "#5213D4",
      "fontFamily": "Luckiest Guy",
      "fontSize": 24,
      "textAlign": "center",
    }
  }>
  challenge 
  Darien
  !
</div>
`;

exports[`Warcry returns null if the player is not challengeable 1`] = `null`;
